### PR TITLE
feat(core): add backward GGSW Fourier conversion to FFT backend

### PR DESCRIPTION
**Pull request by agnesLeroy**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [x] The draft release description has been updated
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs [hunk 1]
```diff
@@ -201,3 +203,129 @@ impl GgswCiphertextDiscardingConversionEngine<GgswCiphertext64, FftFourierGgswCi
...
+        let fft = fft.as_view();
+        self.resize(
+            fill_with_backward_fourier_scratch(fft)
+                .unwrap()
+                .unaligned_bytes_required(),
+        );
+        input
+            .0
+            .as_view()
+            .fill_with_backward_fourier(output.0.clone(), fft, self.stack());
```
Comment (sarah-ek) [inline concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs:236]: here we're modifying a clone of the output, and not the output itself. `fill_with_backward_fourier` should be changed to take a mutable slice type, and this should pass to it `output.0.as_mut_view()`

File: concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs [hunk 2]
```diff
@@ -201,3 +203,129 @@ impl GgswCiphertextDiscardingConversionEngine<GgswCiphertext64, FftFourierGgswCi
...
```
Comment (sarah-ek) [inline concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs:329]: same comment here. we're modifying a clone of the output, and not the output itself

File: concrete-core/src/backends/fft/private/crypto/ggsw.rs [hunk 3]
```diff
@@ -283,6 +284,41 @@ impl<'a> FourierGgswCiphertextMutView<'a> {
...
+pub fn fill_with_backward_fourier_scratch(fft: FftView<'_>) -> Result<StackReq, SizeOverflow> {
+    fft.backward_scratch()
+}
+
+impl<'a> FourierGgswCiphertextView<'a> {
+    /// Fills a standard GGSW ciphertext with the inverse Fourier transform of a GGSW ciphertext
+    /// in the Fourier domain.
+    pub fn fill_with_backward_fourier<Scalar: UnsignedTorus>(
+        self,
+        mut coef_ggsw: StandardGgswCiphertext<Vec<Scalar>>,
```
Comment (sarah-ek) [inline concrete-core/src/backends/fft/private/crypto/ggsw.rs:297]: `coef_ggsw`'s type should be `StandardGgswCiphertext<&mut [Scalar]>`. that way we can modify the output through the view that we receive.

**Review by sarah-ek — CHANGES_REQUESTED** [review]

File: concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs (hunk 1 above)
Comment (agnesLeroy) [reply concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs]: Oh sure, just fixed it! Thank you :slightly_smiling_face:

File: concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs (hunk 2 above)
Comment (agnesLeroy) [reply concrete-core/src/backends/fft/implementation/engines/fft_engine/ggsw_ciphertext_discarding_conversion.rs]: Fixed

File: concrete-core/src/backends/fft/private/crypto/ggsw.rs (hunk 3 above)
Comment (agnesLeroy) [reply concrete-core/src/backends/fft/private/crypto/ggsw.rs]: Fixed

**Review by sarah-ek — APPROVED** [review]

**Pull request merged**